File: src/lib/static-server.ts
/**
 * Frontend wrapper for the built-in static file server and project type detection.
 *
 * Used for plain HTML/CSS/JS projects that don't have a framework dev server.
 *
 * @module lib/static-server
 */

import { invoke } from '@tauri-apps/api/core';

/** Detected project type from the Rust backend */
export type ProjectType = 'nextjs' | 'sveltekit' | 'astro' | 'nuxt' | 'statichtml' | 'unknown';

/** Detect the project type for a given project path */
export async function detectProjectType(projectPath: string): Promise<ProjectType> {
  return invoke<ProjectType>('detect_project_type_command', { projectPath });
}

/** Start the built-in static file server, returns the port it's listening on */
export async function startStaticServer(windowLabel: string, projectPath: string): Promise<number> {
  return invoke<number>('start_static_server', { windowLabel, projectPath });
}

/** Stop the static file server for a window */
export async function stopStaticServer(windowLabel: string): Promise<void> {
  return invoke<void>('stop_static_server', { windowLabel });
}

File: src-tauri/src/commands/static_server.rs
//! # Static Server Commands
//!
//! Tauri commands for managing the built-in static file server
//! used by plain HTML/CSS/JS projects.

use crate::utils::validate_project_path;

/// Start a static file server for a project, returning the port it's listening on.
#[tauri::command]
pub async fn start_static_server(
    window_label: String,
    project_path: String,
) -> Result<u16, String> {
    let validated = validate_project_path(&project_path)?;
    crate::static_server::start_static_server(window_label, validated.to_string_lossy().to_string())
        .await
}

/// Stop the static file server for a window.
#[tauri::command]
pub fn stop_static_server(window_label: String) -> Result<(), String> {
    crate::static_server::stop_static_server(&window_label);
    Ok(())
}
